### ISSUE TITLE
After upgrade catboost from 0.21 to 0.24.4 the learning a model with `"loss_function": "RMSE"` was broken

### ISSUE THREAD

**Issue by boris-pichugin** [catboost/catboost#1610]
# Problem
After upgrade catboost from 0.21 to 0.24.4 the learning a model with `"loss_function": "RMSE"` was broken.

Mean AUC after 30 learnings computed by the learn dataset:
| trees | 0.21 | 0.24.4 |
| -- | -- | -- |
| 200 | 0.7823 | 0.7795 |
| 1000 | 0.8105 | 0.7965 |
| 3000 | 0.8245 | 0.7975 |
| 10000 | 0.8432 | 0.7975 |

Mean AUC after 30 learnings computed by the test dataset:
| trees | 0.21 | 0.24.4 |
| -- | -- | -- |
| 200 | 0.7702 | 0.7667 |
| 1000 | 0.7976 | 0.7834 |
| 3000 | 0.8074 | 0.7841 |
| 10000 | 0.8143 | 0.7841 |

# Parameters of learning:
```
{
  "iterations": 10000,
  "loss_function": "RMSE",
  "metric_period": 100,
  "learning_rate": 0.04,
  "depth": 6,
  "task_type": "GPU",
  "max_ctr_complexity": 1,
  "devices": "0-7",
  "custom_metric": [],
  "nan_mode": "Min",
  "use_best_model": false,
  "random_seed": 42,
  "used_ram_limit": "129928180kb",
  "thread_count": 32,
  "gpu_ram_part": 0.98,
  "verbose": 100,
  "train_dir": "model/train000-0"
}
```

# catboost version
`0.21`, `0.24.4`

# Operating System
```
/opt/python/cb24/bin# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.2 LTS
Release:        18.04
Codename:       bionic
```

# CPU
`Intel(R) Xeon(R) Gold 6151 CPU @ 3.00GHz`

# GPU
```
/opt/python/cb24/bin# nvidia-smi
Mon Mar 15 19:30:18 2021
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 450.51.06    Driver Version: 450.51.06    CUDA Version: 11.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla V100-PCIE...  On   | 00000000:00:0D.0 Off |                  Off |
| N/A   50C    P0   132W / 250W |  15845MiB / 16160MiB |     90%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
```

# More details

## Dataset shape

Learn shape: `(1080403, 519)`, size: 2.32 GB
Test shape: `(363248, 519)`, size: 0.78 GB

## Target labels distribution

```
Labels: min = 0.0023836439693527035, max = 0.9999999761581421
(0.00038999999999999994, 0.102]    803584
(0.102, 0.202]                          0
(0.202, 0.302]                     178544
(0.302, 0.401]                      11296
(0.401, 0.501]                      32518
(0.501, 0.601]                     354770
(0.601, 0.701]                          0
(0.701, 0.8]                        44630
(0.8, 0.9]                              0
(0.9, 1.0]                          18309
```

## Training code

```Python
def _dataframe_to_pool(features: Columns, df: pd.DataFrame, label_column: str, group_by: str) -> Pool:
    return Pool(
        data=df[[c.full_name for c in features if c.is_feature()]],
        label=df[label_column],
        group_id=df[group_by],
        weight=df["weight"] if "weight" in df.columns else None,
        cat_features=[c.full_name for c in features if c.is_categorical_feature()]
    )

learn_pool = _dataframe_to_pool(features, learn_df, "label", group_by)
test_pool = _dataframe_to_pool(features, test_df, "label", group_by)
model = CatBoost(catboost_params).fit(X=learn_pool, eval_set=test_pool)
```

---

I hereby agree to the terms of the CLA available at: [https://yandex.ru/legal/cla/?lang=ru].

**Comment by kizill** [comment]
Hello! I guess main change that could affect GPU in this case was switching from ```bootstrap_type='Bayesian'``` to ```bootstrap_type='MVS'``` by default. Can you try setting ```bootstrap_type='Bayesian'```?

**Comment by boris-pichugin** [comment]
Thank you. I'll try to do it.

ср, 17 Мар 2021 г., 22:02 Stanislav Kirillov ***@***.***>:

**Comment by boris-pichugin** [comment]
Tank you very much!

`bootstrap_type='Bayesian'` is worked for me.
